File: BaekJoon/1764_듣보잡.kt
fun main() {
    val (n,m) = readLine()!!.split(" ").map { it.toInt() }
    val words_n = mutableMapOf<String,Int>()
    for (i in 0..n-1){
        words_n.put(readLine().toString(),1)
    }
    var result_list = mutableListOf<String>()
    for (i in 0..m-1){
        val word = readLine().toString()
        if (word in words_n){
            result_list.add(word)
        }
    }
    result_list.sort()
    println(result_list.size)
    for (crnt in result_list){
        println(crnt)
    }
}

File: BaekJoon/1764_듣보잡.py
from sys import stdin

n,m = map(int, stdin.readline().split(" "))
words_n = {stdin.readline().strip():1 for _ in range(n)}
result_arr = []
for _ in range(m):
    word = stdin.readline().strip()
    if word in words_n:
        result_arr.append(word)
result_arr.sort()

print(len(result_arr))
for crnt in result_arr:
    print(crnt)
